### ISSUE TITLE
Using swipes in TextInput to scroll on Android

### ISSUE THREAD

**Issue by Alspb** [kivy/kivy#7320]
**Feature request**

Currently in TextInput swipes (i.e. on_touch_move) are used to select text. That's usual and convenient for desktop OS, but on Andorid swipes are typicaly used to scroll text (up/down - for multiline, left/right - for single-line).
Additionally, there is only one way to scroll text on Android (without selecting it) - by moving middle handle, and this option is not really convenient.

So it seems reasonable to use swipes in TextInput for scrolling if use_handles is True, and for text selection if use_handles is False (because without handles one need some way to select text).
It'd also be great to add scroll bar and some scrolling effects, but it's not too important.

There is a way to use ScrollView:
```
ScrollView:
    id: scrlv
    do_scroll_x: False
    do_scroll_y: True
    TextInput:
        size_hint: 1, None
        use_handles: True
        height: max(self.minimum_height, scrlv.height)
```
But there are several problems here:
1) On Android app stops responding for a really huge text (like '\n'.join(['abcd{}'.format(i) for i in range(10000)])).
2) Moving cursor outside of the visible area (with keyboard arrows or middle handle) doesn't perform scrolling to the cursor.

**Comment by Alspb** [comment]
ScrollView solution above doesn't work because of the huge TextInput height (which causes the Memory error even on desktop and also problems with scrolling to cursor).
Haven't found a way to deal with this problem yet. 

Также лучше дописать, что вариант со ScrollView при большем объеме текста на Linux тоже ломается - MemoryError. Дело в большой height (которая определяется по minimum_height). При ряд методов TextInput также использует self.height, из-за чего курсор и уходит из области видимости (предположительно проблема в методе _set_cursor).

**Comment by misl6** [comment]
Fixed via #7610
